lib/cluster_ht_item.rb

```diff
@@ -98,17 +101,39 @@ def update_or_add_ht_items(cluster)
     Reclusterer.new(cluster).recluster if needs_reclustering
   end
 
-  def needs_recluster?(cluster, current_ocns, new_ocns)
+  def update_needs_recluster?(cluster, old_ocns, new_ocns)
     # Before updating the HTItem, we will have already merged all the ocns in
     # new_ocns into the cluster. We only need to recluster (i.e. potentially
-    # split) if current_ocns has OCNs that are in NEITHER the new_ocns NOR the
-    # concordance rules.
-
-    current_ocns = current_ocns.to_set
+    # split) if the item could have been the 'glue' holding multiple OCNs
+    # together. The following situations mean an HTItem cannot be glue, so we
+    # don't need to recluster:
+    #
+    # - There was 0 or 1 old OCN (so it couldn't have been 'glue')
+    # - old_ocns are all in concordance rules (so this item is not the 'glue')
+    # - old_ocns are a subset of the new ocns (so it remains 'glue')
+
+    old_ocns = old_ocns.to_set
     new_ocns = new_ocns.to_set
     concordance_ocns = cluster.ocn_resolutions.collect(&:ocns).flatten.to_set
 
-    !(current_ocns.subset?(new_ocns) || current_ocns.subset?(concordance_ocns))
+    !(old_ocns.count <= 1 ||
```

Comment:
Worth pondering more whether we should extract the common logic here and with `move_needs_recluster?` - is there a named concept here we can extract or is the similarity incidental?

Reply:
I *think* this is redundant. `move_needs_cluster?` is simply `update_needs_recluster?` with `new_ocns` as an empty set. In that case `old_ocns.subset?(new_ocns)` will always be false.

Changing `update_needs_recluster?(cluster, old_ocns, new_ocns=[])` and aliasing `move_needs_recluster?` got no complaints from rspec.